models/ClimateSensor.json

```diff
@@ -0,0 +1,86 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:azsamples:iotdd:ClimateSensor;1",
+  "@type": "Interface",
+  "displayName": "Climate Sensor",
+  "description": "Provides functionality to report temperature, humidity, Pressure",
+  "comment": "Requires temperature, hunidity and pressure sensors.",
+  "contents": [
+    {
+      "@type": "Property",
+      "displayName": "Update interval",
+      "description": "Interval of telemetry updates in seconds",
+      "name": "updateInterval",
+      "writable" : true,
+      "schema": "integer"
+    },
+    {
+      "@type": [
+        "Telemetry",
+        "SemanticType/Temperature"
+      ],
+      "description": "Current temperature on the device",
+      "displayName": "Temperature",
+      "name": "temperature",
+      "schema": "double"
+    },
```

Comment:
The semantic type Temperature requires a "unit" property, so we should add that here.

Reply:
I tried to add a unit like `"unit": "degreeCelsius"` but when validating with the parser got:

```
Element dtmi:azsamples:iotdd:ClimateSensor:_contents:_humidity;1 path unit has value gramPerCubicMetre%E2%80%AF, which violates class constraint value Unit
```

Reply:
It looks like your "unit": "gramPerCubicMetre" has a space in it. If you remove the space, then it parses with the DTDL v2 parser.

---
In reply to: [417696281](https://github.com/rido-min/pnp-ppr-sample-spec/pull/2#discussion_r417696281) [](ancestors = 417696281)